### DISCUSSION TITLE
Getting wrong media url on subdomain

### DISCUSSION THREAD

**Discussion by mmtaleghani** [spatie/laravel-medialibrary#1913] [Q&A] [answered]
By using package on subdomain routes of laravel i'm getting the wrong media url.
e.g image on the main site is:  
http://test.com:8888/storage/1/1912790_7fb7.jpg
by using on the subdomain i get something like this:
http://subdomain.test.com:8888/**test**/storage/1/1912790_7fb7.jpg

steps for reproduce :
using spatie media library version 7.0.0
use spatie media library in subdomain based laravel project.
                                                                           
I would be appreciate if anyone could help with this problem.

**Comment by miclf** [comment, marked as answer]
You can modify how the package generates URLs by creating a custom class to override the default behaviour. [This documentation page](https://docs.spatie.be/laravel-medialibrary/v7/advanced-usage/generating-custom-urls/) for version 7 explains that (it works globally the same in version 8).

It seems this is the best way to adapt the package to your own use case.
